projects/plugins/boost/app/assets/src/js/features/critical-css/critical-css-meta/critical-css-meta.tsx

```diff
@@ -34,6 +34,14 @@ const CriticalCssMeta: React.FC< CriticalCssMetaProps > = ( { isCloudCssAvailabl
 
 	const { progress } = useLocalCriticalCssGenerator();
 
+	// On mount, check for not_generated status, and request generation if needed.
+	useEffect( () => {
+		if ( cssState.status === 'not_generated' ) {
+			regenerateAction.mutate();
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps -- Only run on mount.
+	}, [] );
+
```

Comment:
I think `onEnable`/`onMountEnabled` might be clearer for this purpose. I don't feel too good about putting the responsibility of generating critical CSS to the meta component.

Reply:
You're right, it doesn't belong there. But I disagree about putting it in `index`; that is bloated enough, and this functionality will get lost there.

I've moved it into the `useLocalCriticalCssGenerator` hook in edc7defc31.

I realized that it's not about running this on mount. It's about making sure the generator auto-runs if it is in a not_generated state when the module is enabled.